static/app/components/searchSyntax/parser.tsx

```diff
@@ -765,8 +765,15 @@ const options = {
 };
 
 /**
- * Parse a search query into a ParseResult
+ * Parse a search query into a ParseResult. Failing to parse the search query
+ * will result in null.
  */
-export function parseSearch(query: string): ParseResult {
-  return grammar.parse(query, options);
+export function parseSearch(query: string): ParseResult | null {
+  try {
+    return grammar.parse(query, options);
+  } catch {
+    // TODO(epurkhiser): In the future we will either want to guarentee that
+    // all search queries parse, or bubble up the error somehow.
+    return null;
```

Comment:
The interface for parseSearch has changed, so errors now produce `null` and we throw away the error message...

Going to toss a sentry transaction in here actually

Reply:
yup we can def instrument this to see number of incorrect searches. Might be noisy though since the grammar can error when people are in the middle of writing something valid

Reply:
Adding now